packages/sdk/src/Message.ts

```diff
@@ -1,8 +1,24 @@
 import { HexString, StreamID } from '@streamr/utils'
-import { SignatureType, StreamMessage } from './protocol/StreamMessage'
+import { StreamMessage } from './protocol/StreamMessage'
+import { SignatureType } from '@streamr/trackerless-network'
+import { identityConfig, KeyType } from './identity/identityConfig'
+
+// Lookup structure for converting SignatureType to KeyType string
+export type MessageSignatureType = KeyType | 'ECDSA_SECP256K1_LEGACY' | 'ERC_1271'
+const stringVersionsOfSignatureTypes: Record<number, MessageSignatureType> = {
+    // Read key pair SignatureTypes from identityConfig
+    ...Object.fromEntries(
+        Object.entries(identityConfig).map(
+            ([keyType, config]) => [config.signatureType, keyType as KeyType]
+        )
+    ),
+    // These special ones need to be added manually
+    [SignatureType.ECDSA_SECP256K1_LEGACY]: 'ECDSA_SECP256K1_LEGACY',
+    [SignatureType.ERC_1271]: 'ERC_1271',
+}
 
 /**
- * Represents a message in the Streamr Network.
+ * Represents a message in the Streamr Network. This is an application-facing class, whereas StreamMessage is considered internal.
```

Comment:
Could move this comment to the `StreamMessage` class. There it would be available for those users who care about the fact that there are actually two different message structures. Typical users may just find this separation confusing as it is not relevant for them.

```suggestion
 * Represents a message in the Streamr Network. 
```